test/integration/executor/get_asset_info_test.cpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "integration/executor/executor_fixture.hpp"
+
+#include <vector>
+
+#include <gtest/gtest.h>
+#include "framework/common_constants.hpp"
+#include "integration/executor/query_permission_test.hpp"
+#include "module/shared_model/mock_objects_factories/mock_command_factory.hpp"
+#include "module/shared_model/mock_objects_factories/mock_query_factory.hpp"
+
+using namespace common_constants;
+using namespace executor_testing;
+using namespace framework::expected;
+using namespace shared_model::interface::types;
+
+using iroha::ametsuchi::QueryExecutorResult;
+using shared_model::interface::AssetResponse;
+using shared_model::interface::permissions::Role;
+
+constexpr PrecisionType kAssetPrecision(1);
+
+struct GetAssetInfoTest : public ExecutorTestBase {
+  void prepareAsset() {
+    SCOPED_TRACE("GetAssetInfoTest::prepareAsset");
+    createAsset(kAssetName, kDomain, kAssetPrecision);
+  }
+
+  /// Check the response.
+  void validateResponse(const AssetResponse &response) {
+    EXPECT_EQ(response.asset().assetId(), kAssetId);
+    EXPECT_EQ(response.asset().domainId(), kDomain);
+    EXPECT_EQ(response.asset().precision(), kAssetPrecision);
+  }
+
+  /// Query asset info.
+  QueryExecutorResult query(AccountIdType command_issuer = kAdminId) {
+    return getItf().executeQuery(
+        *getItf().getMockQueryFactory()->constructGetAssetInfo(kAssetId),
+        command_issuer);
+  }
+};
+
+using GetAssetInfoBasicTest = BasicExecutorTest<GetAssetInfoTest>;
+
+/**
+ * @given a user with all related permissions
+ * @when GetAssetInfo is queried on a nonexistent asset
+ * @then there is an error
+ */
+TEST_P(GetAssetInfoBasicTest, InvalidNoAsset) {
+  checkQueryError<shared_model::interface::NoAssetErrorResponse>(
+      query(), error_codes::kNoStatefulError);
+}
+
+INSTANTIATE_TEST_CASE_P(Base,
+                        GetAssetInfoBasicTest,
+                        executor_testing::getExecutorTestParams(),
+                        executor_testing::paramToString);
+
+using GetAssetInfoPermissionTest =
+    query_permission_test::QueryPermissionTest<GetAssetInfoTest>;
+
+TEST_P(GetAssetInfoPermissionTest, QueryPermissionTest) {
+  ASSERT_NO_FATAL_FAILURE(prepareState({}));
+  prepareAsset();
+  checkResponse<AssetResponse>(query(getSpectator()),
+                               [this](const AssetResponse &response) {
+                                 this->validateResponse(response);
+                               });
+}
+
+INSTANTIATE_TEST_CASE_P(Common,
+                        GetAssetInfoPermissionTest,
+                        query_permission_test::getParams({boost::none},
+                                                         {boost::none},
```

Comment:
```suggestion
                                                         {},
```

Reply:
I suggest to revert the change in `getParams`, because `RolePermissionSet` argument is more flexible then optional to a single permission, and the code is simpler.

Reply:
here the optional brings another behaviour: it shows that there is no such permission set. delegating this meaning to an empty permission set can be more resource-cheap, but IMHO does not worth readability. current variant is easier to understand, because in it when we have a permission set, we test it, and when we do not, we do not test, thus making no  assumptions about whether it can be empty or not.